src/lib/proof_cache_tag/lmdb/proof_cache_tag.ml

```diff
@@ -0,0 +1,44 @@
+open Async
+open Core
+module Cache = Disk_cache.Make (Pickles.Proof.Proofs_verified_2.Stable.Latest)
+
+type cache_db = Cache.t
+
+type t = { cache_id : Cache.id; cache_db : cache_db }
+
+let create_db = Cache.initialize
+
+let unwrap t = Cache.get t.cache_db t.cache_id
+
+let generate cache_db proof =
+  let cache_id = Cache.put cache_db proof in
+  { cache_id; cache_db }
+
+module For_tests = struct
+  let create_db () =
+    let open Deferred.Let_syntax in
+    Thread_safe.block_on_async_exn (fun () ->
```

Comment:
I suggest that we implement the `For_tests` in a slightly different way.

Let's update some stuff above:

```
type cache_db = Lmdb of Cache.t | Identity_db

type t = Lmdb of { cache_id : Cache.id; cache_db : cache_db } | Identity of Mina_base.Proof.t

let unwrap = function
  | Lmdb t -> Cache.get t.cache_db t.cache_id
  | Identity proof -> proof

let generate db proof =
  match db with | Lmdb cache_db -> { cache_id = = Cache.put cache_db proof; cache_db } | Identity_db -> Idenity proof
```

Then:

```
module For_tests = struct
   let create_db () = Identity_db
```

Reply:
I've implemented it in similar way in new commits. However, I have doubts if readability does not suffer ..

Reply:
It does suffer a bit, but I think it's better this way for use in tests. We normally want to limit number of components being tested in unit tests. I.e. it seems wasteful to initialize a temporary directory to use for cached proofs for cases when we are not so interested in testing proof caching, when we're testing some code that is completely unrelated to that mechanism.